File: api-finder/src/main/java/br/com/boraviajar/apifinder/document/Travel.java
package br.com.boraviajar.apifinder.document;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class Travel implements Serializable {

    private static final long serialVersionUID = 1L;

    @JsonProperty("destino")
    private String destination;

    @JsonProperty("descricao")
    private String description;

    @JsonProperty("cidadeOrigem")
    private String departureCity;

    @JsonProperty("dataPartida")
    private LocalDate departureDate;

    @JsonProperty("dataRetorno")
    private LocalDate returnDate;

    @JsonProperty("horarioSaida")
    private LocalTime departureTime;

    @JsonProperty("horarioRetorno")
    private LocalTime returnTime;

    @JsonProperty("preco")
    private BigDecimal price;

    @JsonIgnore
    private String agencyId;
}

File: api-watcher/src/main/kotlin/br/com/boraviajar/apiwatcher/listener/model/Travel.kt
package br.com.boraviajar.apiwatcher.listener.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalTime

data class Travel(

        @JsonProperty("id")
        val id: String,

        @JsonProperty("destino")
        val destination: String,

        @JsonProperty("cidadeOrigem")
        val departureCity: String,

        @JsonProperty("descricao")
        val description: String,

        @JsonProperty("dataPartida")
        val departureDate: LocalDate,

        @JsonProperty("dataRetorno")
        val returnDate: LocalDate,

        @JsonProperty("horarioSaida")
        val departureTime: LocalTime,

        @JsonProperty("horarioRetorno")
        val returnTime: LocalTime,

        @JsonProperty("preco")
        val price: BigDecimal
)